exams/pearson/download.py

```diff
@@ -0,0 +1,248 @@
+"""Pearson SFTP download implementation"""
+import logging
+import os
+import zipfile
+from contextlib import contextmanager
+
+from django.conf import settings
+from paramiko import SSHException
+
+from exams.pearson.constants import (
+    EAC_SUCCESS_STATUS,
+    PEARSON_FILE_TYPE_EAC,
+    PEARSON_FILE_TYPE_VCDC,
+    VCDC_SUCCESS_STATUS,
+)
+from exams.pearson.exceptions import RetryableSFTPException
+from exams.pearson.readers import (
+    EACReader,
+    VCDCReader,
+)
+from exams.pearson import utils
+from exams.models import (
+    ExamAuthorization,
+    ExamProfile,
+)
+
+log = logging.getLogger(__name__)
+
+
+@contextmanager
+def locally_extracted(zip_file, member):
+    """
+    Context manager for temporarily extracting a zip file member to the local filesystem
+
+    Args:
+        zip_file (zipfile.ZipFile): the zip file to extract from
+        member (str|zipfile.ZipInfo): the name or ZipInfo of the member to extract
+
+    Yields:
+        file: the extracted file object
+    """
+    extracted_file_path = zip_file.extract(member, path=settings.EXAMS_SFTP_TEMP_DIR)
+    try:
+        with open(extracted_file_path, 'r') as extracted_file:
```

Comment:
Ok, so here's the deal, `csv.DictReader` can take an str or an object that responds to the iterator protocol and returns strings. Opening in binary mode makes the file object's iterator return bytes so I'd need to add some intermediate code  like `data = [str(line) for line in extracted_file]` which to me makes the code a bit messier. The file should always be text so I don't see an issue opening the file in this way anyway.